dashboard/test/controllers/levels_controller_test.rb

```diff
@@ -298,6 +305,36 @@ class LevelsControllerTest < ActionController::TestCase
     assert_response :success
     level = assigns(:level)
     assert_equal level.properties[:toolbox_blocks.to_s], @program
+    assert_nil level.properties[:solution_image_url.to_s]
+  end
+
+  test "should update solution image when updating solution blocks" do
+    post :update_blocks, params: {
+      level_id: @level.id,
+      game_id: @level.game.id,
+      type: 'solution_blocks',
+      program: @program,
+      image: 'stub-image-data',
+    }
+    assert_response :success
+    level = assigns(:level)
+    puts "level.properties #{level.properties}"
+    assert_equal level.properties[:solution_blocks.to_s], @program
+    assert_s3_image_url(level.properties[:solution_image_url.to_s])
+  end
+
+  test "should not update solution image when updating toolbox blocks" do
+    post :update_blocks, params: {
+      level_id: @level.id,
+      game_id: @level.game.id,
+      type: 'toolbox_blocks',
+      program: @program,
+      image: 'stub-image-data',
```

Comment:
These parameters appear to be (mostly) duplicated across tests. It might make sense to define a `default_params` in the setup block, then modify it appropriately for each test. In addition to being DRY, it makes it more clear what part of the param is under test.